### PR TITLE
Add `save-images` option for ignoring a number of frames at start/end.

### PR DESCRIPTION
**Pull request by tonycpsu**
Add `--image-frame-margin` option, which specifies a number of frames
to ignore at the beginning and end of scenes when generating images.

This is useful if scene cuts may have frames from the previous/next
scene in them for some reason, or if there are some transition effects
or camera movement between scenes.

### CODE REVIEW

File: scenedetect/cli/context.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def _generate_images(self, scene_list, video_name,
         else:
             middle_images = self.num_images - 2
             for i, (start_time, end_time) in enumerate(scene_list):
-                timecode_list[i].append(start_time)
+                timecode_list[i].append(start_time + self.image_frame_margin)
```
Comment (Breakthrough) [inline scenedetect/cli/context.py]: Would prefer bound checks on the timecodes being used to make sure they do not exceed the frame boundaries.  Specifically, if the calculated start/end times would exceed the half-way point, I think that should be defined behavior - what that behavior should be, though, I'm not 100% sure on, so any feedback or opinions you have would be welcome.

Comment (Breakthrough) [reply scenedetect/cli/context.py]: Also I know I'm essentially asking you to do work for free, so if you are unable to support any of the requested changes I've made on this PR or any other ones, please don't hesitate to let me know, and I'll take over the rest of the PR accordingly.

Very pleased with the quality of your commits thus far though (very small and targeted changes), sorry I don't have more time to actively support them as much as I used to.  Thank you again for your PRs and issue requests, these are definitely things I want to add in v0.5.2 if possible (hurray for non-breaking changes :))!

Comment (tonycpsu) [reply scenedetect/cli/context.py]: See what you think about a393efc.   It's a pretty significant rewrite of the code that builds the timecode list for images, but I think it accomplishes what you want by dividing each scene's frames into `--num-images` chunks and then selecting the appropriate frame from each.

The `min` and `max` make sure that the frame for the first and last images never go past the frames for the middle images.  Lightly tested with some crazy high values of `--image-frame-margin` and it seems to work as expected.

Comment (Breakthrough) [reply scenedetect/cli/context.py]: Awesome, have you tested with values of 0 and 1 by any chance?  Also I'd like to write some unit tests for this eventually as well, but won't let that keep this from being merged.

Would like to do some testing on this myself before merging, but at a first glance your implementation looks a lot cleaner than mine, thank you! 😄 👍

**Comment by Breakthrough** [comment]
After re-reviewing the changes, this will likely be merged as-is, with unit tests being written as part of #88.  Thanks again for the PR, will get this merged by end of the week!

File: scenedetect/cli/context.py [hunk 2]
```diff
@@ -190,43 +193,52 @@ def _generate_images(self, scene_list, video_name,
...
                 self.video_manager.seek(image_timecode)
                 self.video_manager.grab()
                 ret_val, frame_im = self.video_manager.retrieve()
                 if ret_val:
                     file_path = '%s.%s' % (filename_template.safe_substitute(
                         VIDEO_NAME=video_name,
                         SCENE_NUMBER=scene_num_format % (i + 1),
-                        IMAGE_NUMBER=image_num_format % (j + 1)),
+                        IMAGE_NUMBER=image_num_format % (j + 1),
+                        FRAME_NUMBER=image_timecode.get_frames()),
```
Comment (Breakthrough) [inline scenedetect/cli/context.py]: Need to add this to the documentation, but I can do that before merging.

**Pull request merged**